checks/apps/deeplearning/pytorch/src/cnn_distr.py

```diff
@@ -0,0 +1,91 @@
+import os
+import random
+import time
+import numpy as np
+import torch
+import torch.distributed as dist
+import torch.nn.functional as F
+import torch.optim as optim
+from torch.nn.parallel import DistributedDataParallel
+from torch.utils.data import DataLoader, Dataset, DistributedSampler
+from torchvision import models
+from pt_distr_env import DistributedEnviron
+
+
+num_warmup_epochs = 2
+num_epochs = 5
+batch_size_per_gpu = 512
+num_iters = 25
+model_name = 'resnet152'
+
+distr_env = DistributedEnviron()
+dist.init_process_group(backend="nccl")
+world_size = dist.get_world_size()
+rank = dist.get_rank()
+device = 0  # distr_env.local_rank  # since CUDA_VISIBLE_DEVICES=$SLURM_LOCALID
```

Comment:
Is this binding defined somewhere else?

Reply:
Indeed. I made some changes last week from tests that I'm running somewhere else. I still have to try them on lumi. I'm making this PR a draft until I test it.